### ISSUE TITLE
[PoP Miner] Transactions aren't processed if Bitcoin blocks are coming in too fast 

### ISSUE THREAD

**Issue by a123b** [VeriBlock/nodecore-releases#88]
It seems like the Bitcoin testnet is currently being spammed with new blocks, there are multiple new blocks every second. This seems to cause the PoP miner to stop submitting most Bitcoin transactions. Only very few of the operations show `Submitted Bitcoin transaction: <txhash>`, most of them are stuck at `action: 'Creating Bitcoin transaction containing PoP publication data'` and seem to never get out of this state.

Also, the log file is being spammed with: `Did not find block <blockhash> in endorsed block context headers`. It also reveals that all operations actually get broadcasted (`Broadcasting tx <txhash>`, `Awaiting confirmation of broadcast of Tx <txhash>`), but `Received pending tx <txhash>` is only printed for the few which don't get stuck.

It seems like the miner only looks for the transactions in the mempool and if they are already included in a block before a mempool update is received, they simply get ignored.

**Comment by timstall** [comment]
@a123b - yes, BTC testnet getting spammed - https://wiki.veriblock.org/index.php?title=Community_Announcements#Bitcoin_TestNet_Difficulty_Bomb. With BTC testnet stabilizing, does this appear closed for you?

**Comment by a123b** [comment]
I certainly didn't see it happen again after the BTC testnet stabilized, but I'm not sure if this issue should already be closed. 
Maybe the reason for the stuck operations really was that the block containing the transaction was received and added before the pending transaction itself was. If this is true, which I don't know how to test, this could also happen with normal block times, albeit very very rarely.
On the testnet, it really doesn't matter, but as soon as you spend real BTC for PoP mining, you'll want that _every single_ PoP operation gets completed after the BTC are spent.
So I would rather have this opened with low priority then closed.

**Comment by a123b** [comment]
This just happened to me again, some observations:
- For every operation that gets into the `WAIT` state, a line like this is printed to to the bitcoinj log:
```
broadcastTransaction: SEEN_PEERS:  TX <txid> seen by 4 peers
broadcastTransaction: <txid> complete
```
- This line is never printed for operations that are stuck in the `TRANSACTION` state even though the transaction itself appears on block explorers, so it seems like this is actually a bitcoinj bug. It just fails to detect that peers accepted the transaction
- After the first operation got stuck, all subsequent operations got stuck, too. After I stopped sending `mine` commands for a few minutes, it started working again without restarting the miner. In this time period, a tBTC block was found. I don't know if this was what fixed it, or if it was just because many peers were faulty and bitcoinj replaced them with new ones in that time period
- I tracked one of the stuck operations until their transaction got included in a block, and contrary to what I believed earlier, this actually made bitcoinj realize that the transaction was in fact completed:
```
broadcastTransaction: TYPE:  TX <txid> seen by 0 peers and mined
broadcastTransaction: <txid> complete
```
- After this, the PoP miner immediately created and sent the PoP transaction to nodecore, so this problem isn't as severe as I initially thought
